### ISSUE TITLE
Executable with CLI for tarball validation

### ISSUE THREAD

**Issue by AndreasAlbert** [HEPData/hepdata-validator#27]
Use case: I have a tarball ready for uploading and I want to check its integrity. That requires me to:
1. untar the archive
2. validate the submission file using the submission file validator
3. validate each individual data file using the data file validator

Is there a utility tool wrapping this? It would be fantastic if one could just do:
```bash
$ hepdata_validate my_tar_ball.tgz
TARBALL IS GOOD
```

**Comment by GraemeWatt** [comment]
Have you seen the [check.py](https://raw.githubusercontent.com/HEPData/hepdata-submission/master/scripts/check.py) script?  It operates on a directory rather than a tarball, i.e. `./check.py directory`.  It assumes that the `hepdata-validator` Python package is installed, otherwise it only checks that the YAML files can be parsed.  This should do what you need apart from untarring the archive before running `./check.py`.

**Comment by alisonrclarke** [comment]
@GraemeWatt said:

> I think the standalone linked check.py should be integrated within the core HEPData validator package, so it can be used either via Python or a CLI, and operate on either directories or tarballs.

**Comment by GraemeWatt** [comment]
@AndreasAlbert, we implemented your suggestion starting in v0.3.0 of the `hepdata-validator` package released in September.  The latest v0.3.2 was released today with some minor bug fixes.  After installing the `hepdata-validator` package (via `pip`) you can use either the command-line tool `hepdata-validate -a my_tar_ball.tgz` or the Python interface:
```python
from hepdata_validator.full_submission_validator import FullSubmissionValidator
full_submission_validator = FullSubmissionValidator()
is_archive_valid = full_submission_validator.validate(archive='my_tar_ball.tgz')
```
See https://hepdata-validator.readthedocs.io for more information.  Hopefully this new functionality will make it easier for you to address HEPData/hepdata_lib#160.
